### PR TITLE
feat/testing: add tests to ensure that code works as intended

### PR DESCRIPTION
**Pull request by mlederbauer**
**TLDR**: What does this mean for developing this code? Nothing really, you get a checkmark that tells you whether the tests "passed" or not. This so-called unit testing is standard practice to check whether the code works as intended, and is not broken at any stage of coding. **When adding a new feature, it would be good to add tests in the directory `tests/` - but it shouldn't be a priority right now.** Mainly, I've been experimenting with github workflows and this is a first attempt :)

----

This pull request introduces several changes aimed at improving the quality and reliability of the NMRcraft codebase. Specifically, it adds a suite of unit tests and a continuous integration (CI) workflow that runs these tests automatically whenever code is pushed to the main branch or a pull request is opened against it.

Unit Tests
The new unit tests are located in the tests/ directory. These tests cover the load_model function in nmrcraft/models/models.py, ensuring that it correctly loads the supported models and raises an error when an unsupported model or keyword argument is provided.

Here's an example of one of the new tests:
```python
def test_load_model_unsupported_model():
    with pytest.raises(ValueError):
        load_model("unsupported_model")
```

This test asserts that a ValueError is raised when we try to load a model named "unsupported_model", which is not supported by the load_model function.

Continuous Integration Workflow
Continuous Integration Workflow
The new CI workflow is defined in .github/workflows/tests.yml. This workflow runs on Ubuntu and performs the following steps:

1. Checks out the code.
2. Sets up Python 3.11.
3. Installs the project's dependencies using Poetry.
4. Runs the unit tests using the make test command.

Here's an excerpt from the workflow file:
```bash
jobs:
  test:
    runs-on: ubuntu-latest
    steps:
      - uses: actions/checkout@v2
      - name: Set up Python
        uses: actions/setup-python@v2
        with:
          python-version: "3.8"
      - name: Install dependencies
        run: |
          pip install poetry
          poetry install
      - name: Run tests
        run: make test
```
With these changes, we can ensure that all tests pass before any code is merged into the main branch, helping to prevent bugs and regressions.

Please review these changes and let me know if you have any questions or feedback.

### CODE REVIEW

**Comment by mlederbauer** [comment]
The consequence of this feature is what you see next to commits:
image

The green checkmark tells us that all tests have passed. If they haven't, a red X pops up!

**Comment by tiaguinho-code** [comment]
Looks good so far! In the future I will try to make this run on archlinux, like the docker so we have maximum compatibility between the docker and the testing environment.

**Comment by mlederbauer** [comment]
thanks @tiaguinho-code !

**Comment by mlederbauer** [comment]
is there a reason for reverting the changes, did it not pass the tests ?

**Comment by tiaguinho-code** [comment]
> is there a reason for reverting the changes, did it not pass the tests ?

If I see correctly I reverted my own accidental push, where I tried to configure the testing os. If I reverted any of your commits that must have been a mistake.

**Comment by mlederbauer** [comment]
No I meant your commit, but no worries! You can perhaps check if something you want to push passes `make test`, then it should be alright to go to push to GitHub. Feel free to overwrite ubuntu with arch if that is what you planned!

**Comment by tiaguinho-code** [comment]
Yeah, will do. I reverted my commit because I was just messing around a bit and the test got stuck, so I guess I must have broken it lol

**Review by kbiniek — APPROVED** [review]

**Comment by kbiniek** [comment]
Looks great to me! Thank you @mlederbauer for implementing the tests and pipeline! I approved the changes and imo we can already merge it if thats alright with you @tiaguinho-code since we have working tests and the switch to arch is not urgent rn

**Review by tiaguinho-code — APPROVED** [review]

**Comment by mlederbauer** [comment]
thx guys! @tiaguinho-code @kbiniek
